src/scalar/dusk.rs

```diff
@@ -24,6 +25,110 @@ impl PartialOrd for Scalar {
     }
 }
 
+/// Random number generator for generating scalars that are uniformly
+/// distributed over the entire field of scalars.
+///
+/// Because scalars take 255 bits for encoding it is difficult to generate
+/// random bit-pattern that ensures to encode a valid scalar.
+/// Wrapping the values that are higher than [`MODULUS`], as done in
+/// [`Self::random`], results in hitting some values more than others, whereas
+/// zeroing out the highest two bits will eliminate some values from the
+/// possible results.
+///
+/// This function achieves a uniform distribution of scalars by using rejection
+/// sampling: random bit-patterns are generated until a valid scalar is found.
+/// The scalar creation is not constant time but that shouldn't be a concern
+/// since no information about the scalar can be gained by knowing the time of
+/// its generation.
+///
+/// ## Example
+///
+/// ```
+/// use rand::rngs::{StdRng, OsRng};
+/// use rand::SeedableRng;
+/// use dusk_bls12_381::{BlsScalar, UniScalarRng};
+/// use ff::Field;
+///
+/// // using a seedable random number generator
+/// let mut rng: UniScalarRng<StdRng> = UniScalarRng::seed_from_u64(0x42);
+/// let _scalar = BlsScalar::random(rng);
+///
+/// // using randomness derived from the os
+/// let mut rng = &mut UniScalarRng::<OsRng>::default();
+/// let _scalar = BlsScalar::random(rng);
+/// ```
+#[derive(Clone, Copy, Debug, Default)]
+pub struct UniScalarRng<R>(R);
```

Comment:
As mentioned on the JubJub PR, I think adding a new RNG will increase type complexity & degradate performance (i.e. we need to copy the numbers into an array, and then read the numbers from the same array to generate a scalar). I suggest just adding it as a function of the scalar